src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -839,11 +1095,29 @@ public static string ConstructPrimaryKeyRoute(RestRequestContext context, Dictio
 
             foreach (string primaryKey in sourceDefinition.PrimaryKey)
             {
-                // get backing column for lookup, previously validated to be non-null
-                sqlMetadataProvider.TryGetExposedColumnName(entityName, primaryKey, out string? pkExposedName);
+                if (!sqlMetadataProvider.TryGetExposedColumnName(entityName, primaryKey, out string? pkExposedName))
+                {
+                    throw new DataApiBuilderException(
+                       message: "Insert/Upsert operation was successful but unexpected error when constructing the response",
+                       statusCode: HttpStatusCode.InternalServerError,
+                       subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedErrorFindingExposedFieldName);
+                }
+
```

Comment:
Thinking through this a bit more. Do we actually want to fail the entire request if there is an issue creating a location header? Shouldn't the worst case scenario be that we exclude any value for the location header? if the update or create operation(s) succeed, we should probably signal that to the user via 200/201. 500 would indicate some other catastrophic issue.

Reply:
My thought was that if we suppress the error and return an empty string, Location header will contain non-empty value for some requests while containing an empty response for rest of the cases.

It makes sense that we can return a 210 or 200 since the execution reaches that point only after the successful insert/update database operations. Updated to return an empty string